src/idl/tests/annotation.c

```diff
@@ -45,6 +45,210 @@ parse_string(uint32_t flags, const char *str, idl_pstate_t **pstatep)
+    {"struct s { @default(\"hello world!\") string str; };", IDL_RETCODE_OK,              true,  IDL_STRING,  &t5},  //default string
+    {"struct s { @default(123) @optional long l; };",        IDL_RETCODE_SEMANTIC_ERROR,  false, IDL_NULL,    NULL}, //mixing default and optional
+    {"struct s { @default long l; };",                       IDL_RETCODE_SEMANTIC_ERROR,  false, IDL_NULL,    NULL}, //misssing parameter
+    {"struct s { @default(123) string str; };",              IDL_RETCODE_SEMANTIC_ERROR,  false, IDL_NULL,    NULL}, //parameter type mismatch (int vs string)
+    {"struct s { @default(\"false\") boolean b; };",         IDL_RETCODE_SEMANTIC_ERROR,  false, IDL_NULL,    NULL}, //parameter type mismatch (string vs bool)
+    {"struct s { @default(123) boolean b; };",               IDL_RETCODE_SEMANTIC_ERROR,  false, IDL_NULL,    NULL}, //parameter type mismatch (int vs bool)
+    {"union u (switch long) {\n"
+     "case 1: @default(123) short s;\n"
+     "case 2: string s; };",                                 IDL_RETCODE_SYNTAX_ERROR,    false, IDL_NULL,    NULL}, //unions do not allow defaults on values
+    /* skipping this test as idl_create_annotation_appl leaks memory if idl_resolve cannot resolve the scoped name (https://github.com/eclipse-cyclonedds/cyclonedds/issues/950)
+      {"@default(e_0) enum e { e_0, e_1, e_2, e_3 };",         IDL_RETCODE_SEMANTIC_ERROR,  false, IDL_NULL,    NULL}  //setting default on enums is done through @default_literal
+    */
+  };
+
+  for (size_t i = 0; i < sizeof(tests)/sizeof(tests[0]); i++) {
+    test_default(tests[i]);
+  }
+
+}
+
+typedef struct enum_default_test {
+  const char *str;
+  idl_retcode_t ret;
+  uint32_t default_index;
+  const char *default_name;
+} enum_default_test_t;
+
+static void test_enum_default(enum_default_test_t test) {
+  idl_pstate_t *pstate = NULL;
+  idl_retcode_t ret = parse_string(IDL_FLAG_ANNOTATIONS, test.str, &pstate);
+  CU_ASSERT_EQUAL(ret, test.ret);
+  if (test.ret == ret
+   && ret == IDL_RETCODE_OK) {
+    CU_ASSERT_PTR_NOT_NULL_FATAL(pstate);
+    assert(pstate);
+    idl_enum_t *e = (idl_enum_t *)pstate->root;
+    CU_ASSERT(idl_is_enum(e));
+    if (idl_is_enum(e)) {
+      assert(e);
+      idl_enumerator_t *en = e->default_enumerator;
+      CU_ASSERT_PTR_NOT_NULL_FATAL(en);
+      if (en) {
+        CU_ASSERT_EQUAL(en->value.value, test.default_index);
+        CU_ASSERT_EQUAL(idl_mask(en), IDL_DEFAULT_ENUMERATOR);
+        CU_ASSERT_STRING_EQUAL(en->name->identifier, test.default_name);
+      }
+
+      IDL_FOREACH(en, e->enumerators) {
+        if (en == e->default_enumerator) {
+          CU_ASSERT_EQUAL(idl_mask(en), IDL_DEFAULT_ENUMERATOR);
+        } else {
+          CU_ASSERT_EQUAL(idl_mask(en), IDL_ENUMERATOR);
+        }
+      }
+    }
+
+    idl_delete_pstate(pstate);
+  }
+}
+
+CU_Test(idl_annotation, enum_default)
```

Comment:
If you can find the time, please rename the testcase to `default_literal`. That way it's directly obvious in the logs/terminal what this test is meant to test (not that it's super hard to guess, but still).

Reply:
Will do.